docker-compose.yml

```diff
@@ -12,6 +12,38 @@ services:
    volumes:
      - postgres:/var/lib/postgresql/data
 
+ minio:
+   image: minio/minio
+   volumes:
+     - minio:/data
+   ports:
+     - "9000:9000"
+   env_file:
+     - .env
+   environment:
+     MINIO_ROOT_USER: minio
+   command: server /data
+   healthcheck:
+     test: ["CMD", "curl", "-f", "http://localhost:9000/minio/health/live"]
+     interval: 30s
+     timeout: 20s
+     retries: 3
+
+ createbucket:
+   image: minio/mc
+   env_file:
+     - .env
+   depends_on:
+     - minio
+   # Add the desired bucket, and set its policy to public
+   entrypoint: >
+     /bin/sh -c "
+     /usr/bin/mc config host add myminio http://minio:9000 minio ${MINIO_ROOT_PASSWORD};
+     /usr/bin/mc mb myminio/${S3_BUCKET_NAME};
+     /usr/bin/mc policy set public myminio/${S3_BUCKET_NAME};
+     exit 0;
+     "
```

Comment:
This is fascinating! I would have never thought to create a service just to run a script like this.

Reply:
This is part of what took me so long - _normally_, you would add this stuff to the `entrypoint` of the container in question itself, rather than having a completely separate definition in the compose file. Unfortunately, minio the server doesn't ship with minio the command line management utility :roll_eyes: so we have to spin up a transient service at the beginning to make sure everything gets set up.